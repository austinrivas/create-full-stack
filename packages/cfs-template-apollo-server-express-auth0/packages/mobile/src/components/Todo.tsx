import { Todo as TodoType, useDeleteTodo, useUpdateTodoMutation } from "common";
import React from "react";
import { StyleSheet } from "react-native";
import { Button, CheckBox, Icon, ListItem } from "react-native-elements";

const styles = StyleSheet.create({
  lineThrough: {
    textDecorationLine: "line-through",
  },
});

interface Props {
  todo: TodoType;
}

export default function Todo({ todo }: Props) {
  const [updateTodo] = useUpdateTodoMutation();
  const [deleteTodo] = useDeleteTodo();
  const onPress = () =>
    updateTodo({
      variables: { id: todo.id, complete: !todo.complete },
    });

  return (
    <ListItem
      onPress={onPress}
      leftElement={<CheckBox onPress={onPress} checked={todo.complete} />}
      title={todo.name}
      titleStyle={todo.complete ? styles.lineThrough : undefined}
      rightElement={
        <Button
          icon={
            <Icon
              name="delete"
              accessibilityLabel="delete"
              onPress={() => deleteTodo({ variables: { id: todo.id } })}
            />
          }
          type="outline"
          raised
        />
      }
    />
  );
}
