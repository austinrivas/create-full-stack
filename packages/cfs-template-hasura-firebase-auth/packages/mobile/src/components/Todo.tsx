import { Todos as TodoType, useDestroyTodo, useUpdateTodo } from "common";
import React from "react";
import { StyleSheet } from "react-native";
import { Button, CheckBox, Icon, ListItem } from "react-native-elements";

const styles = StyleSheet.create({
  lineThrough: {
    textDecorationLine: "line-through",
  },
});

interface Props {
  todo: TodoType;
}

export default function Todo({ todo }: Props) {
  const [updateTodo] = useUpdateTodo();
  const [destroyTodo] = useDestroyTodo();
  const onPress = () =>
    updateTodo({
      variables: { id: todo.id, complete: !todo.complete },
    });

  return (
    <ListItem
      onPress={onPress}
      leftElement={<CheckBox onPress={onPress} checked={todo.complete} />}
      title={todo.name}
      titleStyle={todo.complete ? styles.lineThrough : undefined}
      rightElement={
        <Button
          icon={
            <Icon
              name="delete"
              onPress={() => destroyTodo({ variables: { id: todo.id } })}
            />
          }
          type="outline"
          raised
        />
      }
    />
  );
}
