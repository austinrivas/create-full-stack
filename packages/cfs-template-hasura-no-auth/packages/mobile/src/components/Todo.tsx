import { Todos as TodoType, useDestroyTodo, useUpdateTodo } from "common";
import React from "react";
import { Button, CheckBox, Icon, ListItem } from "react-native-elements";

interface Props {
  todo: TodoType;
}

export default function Todo({ todo }: Props) {
  const [updateTodo] = useUpdateTodo();
  const [destroyTodo] = useDestroyTodo();

  return (
    <ListItem
      leftElement={
        <CheckBox
          onPress={() =>
            updateTodo({
              variables: { id: todo.id, complete: !todo.complete },
            })
          }
          checked={todo.complete}
        />
      }
      title={todo.name}
      titleStyle={
        todo.complete
          ? {
              textDecorationLine: "line-through",
            }
          : undefined
      }
      rightElement={
        <Button
          icon={
            <Icon
              name="delete"
              onPress={() => destroyTodo({ variables: { id: todo.id } })}
            />
          }
          type="outline"
          raised
        />
      }
    />
  );
}
