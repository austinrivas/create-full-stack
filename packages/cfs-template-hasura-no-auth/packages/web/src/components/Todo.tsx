import {
  Checkbox,
  IconButton,
  ListItem,
  ListItemSecondaryAction,
  ListItemText,
  makeStyles,
} from "@material-ui/core";
import DeleteIcon from "@material-ui/icons/Delete";
import { Todos as TodoType, useDeleteTodo, useUpdateTodo } from "common";
import React from "react";

const useStyles = makeStyles({
  complete: {
    textDecoration: "line-through",
  },
});

interface Props {
  todo: TodoType;
}

export default function Todo({ todo }: Props) {
  const classes = useStyles();
  const [updateTodo] = useUpdateTodo();
  const [destroyTodo] = useDeleteTodo();

  return (
    <ListItem
      key={todo.id}
      role={undefined}
      dense
      button
      onClick={() =>
        updateTodo({ variables: { id: todo.id, complete: !todo.complete } })
      }
    >
      <Checkbox checked={todo.complete} tabIndex={-1} disableRipple />
      <ListItemText
        primary={todo.name}
        classes={todo.complete ? { primary: classes.complete } : undefined}
      />
      <ListItemSecondaryAction>
        <IconButton
          aria-label="Delete"
          onClick={() => destroyTodo({ variables: { id: todo.id } })}
        >
          <DeleteIcon />
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  );
}
