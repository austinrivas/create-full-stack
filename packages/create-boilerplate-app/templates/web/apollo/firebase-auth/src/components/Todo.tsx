import React from "react";
import {
  ListItem,
  Checkbox,
  ListItemText,
  ListItemSecondaryAction,
  IconButton
} from "@material-ui/core";
import DeleteIcon from "@material-ui/icons/Delete";
import {
  Todo as TodoType,
  UpdateTodoComponent,
  TodosQuery,
  TodosQueryVariables,
  TodosDocument,
  DestroyTodoComponent
} from "../generated/graphql";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles({
  complete: {
    textDecoration: "line-through"
  }
});

interface Props {
  todo: TodoType;
}

export default function Todo({ todo }: Props) {
  const classes = useStyles();
  return (
    <UpdateTodoComponent
      update={(cache, { data }) => {
        if (!data) {
          return;
        }
        const updateTodo = data.updateTodo;
        const query = cache.readQuery<TodosQuery, TodosQueryVariables>({
          query: TodosDocument
        });
        if (query) {
          const { todos } = query;
          cache.writeQuery<TodosQuery, TodosQueryVariables>({
            query: TodosDocument,
            data: {
              todos: todos.map(todo =>
                todo.id === updateTodo.id ? updateTodo : todo
              )
            }
          });
        }
      }}
    >
      {updateTodo => (
        <ListItem
          key={todo.id}
          role={undefined}
          dense
          button
          onClick={() =>
            updateTodo({ variables: { id: todo.id, complete: !todo.complete } })
          }
        >
          <Checkbox checked={todo.complete} tabIndex={-1} disableRipple />
          <ListItemText
            primary={todo.name}
            classes={todo.complete ? { primary: classes.complete } : undefined}
          />
          <ListItemSecondaryAction>
            <DestroyTodoComponent
              update={(cache, { data }) => {
                if (!data) {
                  return;
                }
                const destroyTodo = data.destroyTodo;
                const query = cache.readQuery<TodosQuery, TodosQueryVariables>({
                  query: TodosDocument
                });
                if (query) {
                  const { todos } = query;
                  cache.writeQuery<TodosQuery, TodosQueryVariables>({
                    query: TodosDocument,
                    data: {
                      todos: todos.filter(todo => todo.id !== destroyTodo.id)
                    }
                  });
                }
              }}
            >
              {destroyTodo => (
                <IconButton
                  aria-label="Delete"
                  onClick={() => destroyTodo({ variables: { id: todo.id } })}
                >
                  <DeleteIcon />
                </IconButton>
              )}
            </DestroyTodoComponent>
          </ListItemSecondaryAction>
        </ListItem>
      )}
    </UpdateTodoComponent>
  );
}
