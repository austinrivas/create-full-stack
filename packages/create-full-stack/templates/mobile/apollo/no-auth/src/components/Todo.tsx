import React from "react";
import {
  Todo as TodoType,
  UpdateTodoComponent,
  TodosQuery,
  TodosQueryVariables,
  TodosDocument,
  DestroyTodoComponent
} from "../generated/graphql";
import {
  Left,
  Text,
  Right,
  ListItem,
  Button,
  Icon,
  CheckBox
} from "native-base";
import { useUpdateTodoMutation, useDestroyTodoMutation } from "common";

interface Props {
  todo: TodoType;
}

export default function Todo({ todo }: Props) {
  const [updateTodo] = useUpdateTodoMutation();
  const [destroyTodo] = useDestroyTodoMutation();

  return (
    <ListItem>
      <Left>
        <CheckBox
          onPress={() =>
            updateTodo({
              variables: { id: todo.id, complete: !todo.complete }
            })
          }
          checked={todo.complete}
          style={{ marginRight: 40 }}
        />
        <Text
          style={
            todo.complete
              ? {
                  textDecorationLine: "line-through"
                }
              : undefined
          }
        >
          {todo.name}
        </Text>
      </Left>
      <Right>
        <Button onPress={() => destroyTodo({ variables: { id: todo.id } })}>
          <Icon active name="trash" />
        </Button>
      </Right>
    </ListItem>
  );
}
