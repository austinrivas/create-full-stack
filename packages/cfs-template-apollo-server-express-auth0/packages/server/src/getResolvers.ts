import { Connection } from "typeorm";

import { Todo } from "./entity/Todo";
import {
  MutationDeleteTodoArgs,
  MutationUpdateTodoArgs,
  Resolvers,
} from "./graphql/__generated__";

export interface DecodedJwt {
  aud: string[];
  azp: string;
  exp: number;
  iat: number;
  iss: string;
  scope: string;
  sub: string;
}

interface Context {
  user: DecodedJwt;
}

export default function getResolover(connection: Connection): Resolvers {
  const todoRepository = connection.getRepository(Todo);
  return {
    Query: {
      todos: (_, __, context: Context) =>
        todoRepository.find({ uid: context.user.sub }),
    },
    Mutation: {
      createTodo: (_, args, context: Context) =>
        todoRepository.save({
          uid: context.user.sub,
          complete: false,
          ...args,
        }),
      updateTodo: async (
        _,
        { id, ...args }: MutationUpdateTodoArgs,
        context: Context
      ) => {
        await todoRepository.update({ id, uid: context.user.sub }, args);
        return todoRepository.findOne(id);
      },
      deleteTodo: async (
        _,
        { id }: MutationDeleteTodoArgs,
        context: Context
      ) => {
        const todo = todoRepository.findOne({ id, uid: context.user.sub });
        await todoRepository.delete(id);
        return todo;
      },
    },
  };
}
