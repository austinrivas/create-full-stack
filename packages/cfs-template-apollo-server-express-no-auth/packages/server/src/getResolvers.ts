import { Connection } from "typeorm";

import { Todo } from "./entity/Todo";
import {
  MutationCreateTodoArgs,
  MutationDeleteTodoArgs,
  MutationUpdateTodoArgs,
  Resolvers,
} from "./graphql/__generated__";

export default function getResolover(connection: Connection): Resolvers {
  const todoRepository = connection.getRepository(Todo);
  return {
    Query: {
      todos: () => todoRepository.find(),
    },
    Mutation: {
      createTodo: (_, args: MutationCreateTodoArgs) =>
        todoRepository.save({ complete: false, ...args }),
      updateTodo: async (_, { id, ...args }: MutationUpdateTodoArgs) => {
        await todoRepository.update(id, args);
        return todoRepository.findOne(id);
      },
      deleteTodo: async (_, { id }: MutationDeleteTodoArgs) => {
        const todo = todoRepository.findOne(id);
        await todoRepository.delete(id);
        return todo;
      },
    },
  };
}
