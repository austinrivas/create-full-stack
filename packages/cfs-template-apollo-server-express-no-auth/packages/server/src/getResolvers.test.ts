import { ApolloServer } from "apollo-server-express";
import { createTestClient } from "apollo-server-testing";
import {
  CreateTodoDocument,
  DeleteTodoDocument,
  TodosDocument,
  UpdateTodoDocument,
} from "common";
import { createConnection, getConnection, Repository } from "typeorm";

import { Todo } from "./entity/Todo";
import getResolvers from "./getResolvers";
import typeDefs from "./graphql/schema";

let server: ApolloServer;
let todoRepository: Repository<Todo>;

beforeEach(async () => {
  const connection = await createConnection({
    type: "sqlite",
    database: ":memory:",
    dropSchema: true,
    entities: [Todo],
    synchronize: true,
    logging: false,
  });
  const resolvers = getResolvers(connection);
  server = new ApolloServer({
    typeDefs,
    resolvers,
    context: () => ({ user: { sub: "1" } }),
  });
  todoRepository = connection.getRepository(Todo);
});

afterEach(() => {
  const conn = getConnection();
  return conn.close();
});

it("fetches empty todos", async () => {
  const { query } = createTestClient(server);
  const result = await query({ query: TodosDocument });
  expect(result).toMatchSnapshot();
});

it("fetches todos", async () => {
  await todoRepository.save({
    name: "This test",
    uid: "1",
    complete: false,
  });
  const { query } = createTestClient(server);
  const result = await query({ query: TodosDocument });
  expect(result).toMatchSnapshot();
});

it("creates a todo", async () => {
  const { mutate } = createTestClient(server);
  const result = await mutate({
    mutation: CreateTodoDocument,
    variables: { name: "This test" },
  });
  expect(result).toMatchSnapshot();
});

it("updates a todo", async () => {
  const todo = await todoRepository.save({
    name: "This test",
    uid: "1",
    complete: false,
  });
  const { mutate } = createTestClient(server);
  const result = await mutate({
    mutation: UpdateTodoDocument,
    variables: { id: todo.id, complete: true },
  });
  expect(result).toMatchSnapshot();
});

it("deletes a todo", async () => {
  const todo = await todoRepository.save({
    name: "This test",
    uid: "1",
    complete: false,
  });
  const { mutate } = createTestClient(server);
  const result = await mutate({
    mutation: DeleteTodoDocument,
    variables: { id: todo.id },
  });
  expect(result).toMatchSnapshot();
});
