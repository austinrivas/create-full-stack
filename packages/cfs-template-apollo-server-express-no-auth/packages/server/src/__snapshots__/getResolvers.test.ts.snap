// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a todo 1`] = `
Object {
  "data": Object {
    "createTodo": Object {
      "complete": false,
      "id": 1,
      "name": "This test",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`deletes a todo 1`] = `
Object {
  "data": Object {
    "deleteTodo": Object {
      "id": 1,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches empty todos 1`] = `
Object {
  "data": Object {
    "todos": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches todos 1`] = `
Object {
  "data": Object {
    "todos": Array [
      Object {
        "complete": false,
        "id": 1,
        "name": "This test",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`updates a todo 1`] = `
Object {
  "data": Object {
    "updateTodo": Object {
      "complete": true,
      "id": 1,
      "name": "This test",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
